File: packages/python/src/endpoints/jobs.py
"""
Jobs API Client
"""

from typing import Dict, List, Optional, Any
import httpx


class JobsClient:
    """Client for jobs endpoints."""

    def __init__(self, base_url: str, client: Optional[httpx.AsyncClient] = None):
        self.base_url = base_url.rstrip('/')
        self.client = client

    async def _request(self, method: str, path: str, **kwargs) -> Any:
        """Make HTTP request."""
        url = self.base_url + path
        if self.client:
            response = await self.client.request(method, url, **kwargs)
            response.raise_for_status()
            return response.json()
        else:
            async with httpx.AsyncClient() as client:
                response = await client.request(method, url, **kwargs)
                response.raise_for_status()
                return response.json()


    async def get_status_job_id(self, job_id: str):
        """GET /status/{job_id}"""
        url = f"/status/{job_id}"
        params = {", ".join(query_dict)}
        data = None
        return await self._request("GET", url, params=params, json=data)


    async def get_dashboard(self, limit: Optional[float] = None, authorization: Optional[str] = None, x_api_key: Optional[str] = None):
        """GET /dashboard"""
        url = f"/dashboard"
        params = {", ".join(query_dict)}
        data = None
        return await self._request("GET", url, params=params, json=data)

File: packages/typescript/src/endpoints/jobs.ts
/**
 * Jobs API Client
 */

import { AxiosInstance } from 'axios'

export class JobsClient {
  private httpClient: AxiosInstance

  constructor(private baseURL: string, httpClient: AxiosInstance) {
    this.httpClient = httpClient
  }

  /**
   * GET /status/{job_id}
   */
  async getStatus{job_id}(job_id: string): Promise<JobStatus> {
        const params = {'job_id: job_id'}
        const data = undefined
        return this.httpClient.get(`/status/{job_id}`, params)
  }
  /**
   * GET /dashboard
   */
  async getDashboard(limit?: number, authorization?: string, x_api_key?: string): Promise<JobSummary> {
        const params = {'limit: limit, authorization: authorization, x_api_key: x_api_key'}
        const data = undefined
        return this.httpClient.get(`/dashboard`, params)
  }
}
